### PR TITLE
[RW-4335][RW-4336][risk=no] Concept Search - Search by domain

### PR DESCRIPTION
**Pull request by dolbeew**
When viewing a specific domain (after clicking a tile on concept homepage), show only the tab for that domain and only search by that domain.

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] I have run and tested this change locally

### CODE REVIEW

File: common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java [hunk 1]
```diff
@@ -130,6 +130,35 @@
...
+              + "                and dc.name = :surveyName), '.%' "
+              + "          ) "
+              + "          and match(c.concept_name, c.concept_code, c.vocabulary_id, c.synonyms) against (:term in boolean mode)"
+              + "      ) a, "
+              + "      (select @curRow \\:= 0, @curType \\:= '') r "
+              + "    order by concept_name, id "
+              + "  ) as x "
+              + "where rank = 1",
+      nativeQuery = true)
+  long countSurveyByName(@Param("term") String term, @Param("surveyName") String surveyName);
```
Comment (freemabd) [inline common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Need to delete countSurveys()

File: common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java [hunk 2]
```diff
@@ -155,6 +184,41 @@
...
+              + "          and match(c.concept_name, c.concept_code, c.vocabulary_id, c.synonyms) against (:term in boolean mode)"
+              + "      ) a, "
+              + "      (select @curRow \\:= 0, @curType \\:= '') r "
+              + "    order by concept_name, id "
+              + "  ) as x "
+              + "where rank = 1 "
+              + "order by id "
+              + "limit :limit offset :offset",
+      nativeQuery = true)
+  List<DbConcept> findSurveysByName(
```
Comment (freemabd) [inline common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: need to delete `findSurveys(@Param("limit") int limit, @Param("offset") int offset)`

Comment (freemabd) [inline common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Also, can we rename this to findSurveysByTermAndName?

File: common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java (hunk 1 above)
Comment (freemabd) [inline common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Also, can we rename this to countSurveyByTermAndName?

Comment (dolbeew) [reply common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Done

Comment (dolbeew) [reply common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Good idea, done.

File: common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java (hunk 2 above)
Comment (dolbeew) [reply common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Done

Comment (dolbeew) [reply common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java]: Done

**Pull request merged**